test/Microsoft.AspNetCore.SignalR.Tests.Utils/ServerFixture.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    public class ServerFixture<TStartup> : IDisposable
+        where TStartup : class
+    {
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+        private IWebHost _host;
+        private IApplicationLifetime _lifetime;
+        private readonly IDisposable _logToken;
+
+        public string WebSocketsUrl => Url.Replace("http", "ws");
+
+        public string Url { get; private set; }
+
+        public ServerFixture()
+        {
+            var testLog = AssemblyTestLog.ForAssembly(typeof(ServerFixture<TStartup>).Assembly);
+            _logToken = testLog.StartTestLog(null, $"{nameof(ServerFixture<TStartup>)}_{typeof(TStartup).Name}", out _loggerFactory, "ServerFixture");
+            _logger = _loggerFactory.CreateLogger<ServerFixture<TStartup>>();
+            Url = "http://localhost:" + GetNextPort();
+
+            StartServer(Url);
+        }
+
+        private void StartServer(string url)
+        {
+            _host = new WebHostBuilder()
+                .ConfigureLogging(builder => builder.AddProvider(new ForwardingLoggerProvider(_loggerFactory)))
+                .UseStartup(typeof(TStartup))
+                .UseKestrel()
+                .UseUrls(url)
+                .UseContentRoot(Directory.GetCurrentDirectory())
+                .Build();
+
+            var t = Task.Run(() => _host.Start());
```

Comment:
I hate that this virus is spreading. We need to file a bug to fix this logic once and for all...

Reply:
This is just copy-pasta, moving the "Common" code that was continuing to grow into a full-on test utilities assembly.